catalog/core/catalog-core-solrcommands/src/test/java/org/codice/ddf/commands/solr/TestBackupCommand.java

```diff
@@ -135,13 +211,276 @@ protected HttpWrapper getHttpClient() {
+
+        // Verify
+        assertThat(consoleOutput.getOutput(),
+                containsString(String.format(
+                        "Backing up collection [%s] to shared location [%s] using backup name [%s_",
+                        DEFAULT_CORE_NAME,
+                        backupCommand.backupLocation,
+                        DEFAULT_CORE_NAME)));
+        assertThat(consoleOutput.getOutput(), containsString("Solr Cloud backup request Id:"));
+
+        String requestId = getRequestId(consoleOutput.getOutput());
+        consoleOutput.reset();
+
+        // Setup BackupCommand for status lookup
+        BackupCommand statusBackupCommand = getStatusBackupCommand(requestId);
+
+        // Perform status lookup
+        statusBackupCommand.doExecute();
+
+        String status = waitForCompletedStatusOrFail(statusBackupCommand, consoleOutput);
+
+        assertThat(status, is(RequestStatusState.COMPLETED.getKey()));
+    }
+
+    @Test
+    public void testGetSolrCloudAsynchronousBackupStatusNoRequestId() throws Exception {
+        // Setup exception expectations
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("Invalid Argument(s). Run solr:backup --help for usage details.");
+
+        // Set system properties
+        setupSystemProperties(SOLR_CLOUD_CLIENT_TYPE);
+
+        // Setup BackupCommand for status lookup
+        BackupCommand statusBackupCommand = getStatusBackupCommand(null);
+
+        statusBackupCommand.doExecute();
+    }
+
+    @Test
+    public void testGetSolrCloudAsynchronousBackupStatusNoStatusOption() throws Exception {
+
+        // Setup exception expectations
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("Invalid Argument(s). Run solr:backup --help for usage details.");
+
+        // Set system properties
+        setupSystemProperties(SOLR_CLOUD_CLIENT_TYPE);
+
+        // Setup BackupCommand (ie. solr:backup -i <request Id>)
+        BackupCommand invalidBackupStatusCommand = getBackupCommand(null, null, false, false, "myRequestId0");
+
+        invalidBackupStatusCommand.doExecute();
+    }
+
+    @Test
+    public void testGetCloudSolrClientNoZkHosts() throws Exception {
+
+        // Set the solr client type system property so that the
+        // BackupCommand knows that it needs to backup solr cloud.
+        setupSolrClientType(SOLR_CLOUD_CLIENT_TYPE);
+
+        // Setup BackupCommand
+        BackupCommand backupCommand = new BackupCommand();
+        backupCommand.backupLocation = getBackupLocation();
+
+        backupCommand.doExecute();
+
+        // Verify
+        assertThat(consoleOutput.getOutput(), containsString(String.format(
+                String.format("Could not determine Zookeeper Hosts. Please verify that the system property %s is configured in %s.",
+                        ZOOKEEPER_HOSTS_PROP, SYSTEM_PROPERTIES_PATH))));
+    }
+
```

Comment:
❗️We're missing tests for the following cases:
* Backup status fails and returns error messages
* `CollectionAdminRequest.requestStatus(requestId).process(client)` throws an `Exception`
* Optimize collection fails and returns error messages
* Optimize collection throws an `Exception`.